app/Http/Controllers/V4DB/MangaController.php

```diff
@@ -731,9 +733,27 @@ public function reviews(Request $request, int $id)
             || $this->isExpired($request, $results)
         ) {
             $page = $request->get('page') ?? 1;
-            $manga = $this->jikan->getMangaReviews(new MangaReviewsRequest($id, $page));
-            $response = \json_decode($this->serializer->serialize($manga, 'json'), true);
+            $sort = $request->get('sort') ?? Constants::REVIEWS_SORT_MOST_VOTED;
+
+            if (!in_array($sort, [Constants::REVIEWS_SORT_MOST_VOTED, Constants::REVIEWS_SORT_NEWEST, Constants::REVIEWS_SORT_OLDEST])) {
+                throw new BadRequestException('Invalid sort for reviews. Please refer to the documentation: https://docs.api.jikan.moe/');
+            }
 
+            $spoilers = $request->get('spoilers') ?? false;
+            $preliminary = $request->get('preliminary') ?? false;
+
+            $manga = $this->jikan
+                ->getMangaReviews(
+                    new MangaReviewsRequest(
+                        $id,
+                        $page,
+                        $sort,
+                        $spoilers,
+                        $preliminary
+                    )
+                );
```

Comment:
This is also repetition in some way, it would be nicer if there would be a common function to make calls on the "jikan" object where the parameters are the same.

Just an example what I'm thinking:

```php
function getReviews($jikan, $mediaType, $id, $page, $sort, $spoilers, $preliminary) {
  $getterFunc = 'get'.ucfirst($mediaType).'Reviews';
  $requestClassName = ucfirst($mediaType).'ReviewsRequest';
  return $jikan->$getterFunc(new $requestClassName($id, $page, $sort, $spoilers, $preliminary));
}
```
Also you can spice the above type constraints and guards. (E.g. certain strings are allowed or restrict things to enums)

The oop thing would be to use the IoC container, and have a service call functions on the jikan object based on the media type you pass into it 😄